arbnode/seq_coordinator.go

```diff
@@ -473,6 +483,15 @@ func (c *SeqCoordinator) updateWithLockout(ctx context.Context, nextChosen strin
 		return c.noRedisError()
 	}
 	// Was, and still is, the active sequencer
+	// Before proceeding, first try deleting finalized messages from redis and setting the finalizedMsgCount key
+	finalized, err := c.sync.GetFinalizedMsgCount(ctx)
+	if err != nil {
+		log.Warn("Error getting finalizedMessageCount from syncMonitor: %w", err)
+	} else if finalized == 0 {
+		log.Warn("SyncMonitor returned zero finalizedMessageCount")
```

Comment:
I was checking for 0 because sequencer coordinator could be enabled without inboxreader/l1reader and l1reader is required to fetch finalizedMsgCount. In cases when inboxreader/l1reader is nil, sync monitor is made to return `0, nil` this meant a reasonable return value to check if we should skip deleting of finalized messages